# `noUnsafeFinally.test.ts`

**DO NOT MODIFY**. This file has been autogenerated. Run `rome test packages/@romejs/js-compiler/lint/rules/js/noUnsafeFinally.test.ts --update-snapshots` to update.

## `disallow unsafe usage of break, continue, throw and return`

### `0`

```

 unknown:7:4 lint/js/noUnsafeFinally ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unsafe usage of JSReturnStatement.

    5 │     throw err;
    6 │   } finally {
  > 7 │     return 1;
      │     ^^^^^^^^^
    8 │   }
    9 │ }

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✖ Found 1 problem

```

### `0: formatted`

```
function greet1() {
	try {
		throw new Error("Try");
	} catch (err) {
		throw err;
	} finally {
		return 1;
	}
}

```

### `1`

```

 unknown:7:4 lint/js/noUnsafeFinally ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unsafe usage of JSBreakStatement.

    5 │     throw err;
    6 │   } finally {
  > 7 │     break;
      │     ^^^^^^
    8 │   }
    9 │ }

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✖ Found 1 problem

```

### `1: formatted`

```
function greet2() {
	try {
		throw new Error("Try");
	} catch (err) {
		throw err;
	} finally {
		break;
	}
}

```

### `2`

```

 unknown:7:4 lint/js/noUnsafeFinally ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unsafe usage of JSContinueStatement.

    5 │     throw err;
    6 │   } finally {
  > 7 │     continue;
      │     ^^^^^^^^^
    8 │   }
    9 │ }

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✖ Found 1 problem

```

### `2: formatted`

```
function greet3() {
	try {
		throw new Error("Try");
	} catch (err) {
		throw err;
	} finally {
		continue;
	}
}

```

### `3`

```

 unknown:7:4 lint/js/noUnsafeFinally ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unsafe usage of JSThrowStatement.

    5 │     throw err;
    6 │   } finally {
  > 7 │     throw new Error("Finally");
      │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    8 │   }
    9 │ }

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✖ Found 1 problem

```

### `3: formatted`

```
function greet4() {
	try {
		throw new Error("Try");
	} catch (err) {
		throw err;
	} finally {
		throw new Error("Finally");
	}
}

```
